ui/app/AppLayouts/Chat/panels/communities/CommunityMintTokensSettingsPanel.qml

```diff
@@ -184,8 +160,56 @@ SettingsPageLayout {
     onPrimaryHeaderButtonClicked: {
         if(root.state == d.initialViewState)
             stackManager.push(d.newTokenViewState, newTokenView, null, StackView.Immediate)
-        if(root.state == d.tokenViewState)
-            d.retryMintClicked()
+
+        if(root.state == d.tokenViewState) {
+            if(d.currentToken) {
+                if(d.isAssetType) {
+                    // Create new asset instance:
+                    temp_.asset.clearAsset()
```

Comment:
I think I'm missing something. Why do we need the temp objects?

Reply:
Well, here it is needed to create a new `TokenObject` instance (`CollectibleObject` or `AssetObject`) bc in the next line there is a stack reset `stackManager.clear(..)` where the current token object is deleted. The temp object is used to easyly create a new copy object by using the `copyXX(..)` function with the needed information. If you have another suggestion to do the copy object in a more straight forward, please, let me know!

Reply:
> Well, here it is needed to create a new TokenObject instance (CollectibleObject or AssetObject) bc in the next line there is a stack reset stackManager.clear(..) where the current token object is deleted. 

Hmm, `d.currentToken` lives independently from the stack, how it could destroy it? :thinking:

Reply:
I think it's due to the [bindings to asset/collectibles property](https://github.com/status-im/status-desktop/blob/d99093372b20a980a6c0a656e030a50d7e17bce2/ui/app/AppLayouts/Chat/panels/communities/CommunityMintTokensSettingsPanel.qml#L602), but for some reason breaking the binding is not working in this case.
@noeliaSD I wonder if [Binding.RestoreNone](https://doc.qt.io/qt-6/qml-qtqml-binding.html#restoreMode-prop) would do the trick

Reply:
Yes, it is bc of the binding @alexjba indicated. The binded object asset or collectible is created inside a view that is deleted when the stack is cleared. However we've discussed offline that we will keep this implementation as it is and in a next step we will create cpp classes for easyly manage these token objects.